packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4497,66 +4497,452 @@ class PositionedDirectional extends StatelessWidget {
   }
 }
 
-/// A widget that displays its children in a one-dimensional array.
-///
-/// The [Flex] widget allows you to control the axis along which the children are
-/// placed (horizontal or vertical). This is referred to as the _main axis_. If
-/// you know the main axis in advance, then consider using a [Row] (if it's
-/// horizontal) or [Column] (if it's vertical) instead, because that will be less
-/// verbose.
+/// A widget that is the basis for [Row] and [Column]. It arranges its
+/// children in a single line – either horizontally (a [Row]) or vertically
+/// (a [Column]).
```

Comment:
Well they are extending [Flex] and I think that's an important point to make sure people understand from the very beginning. What about stating that instead of saying it's a more generalized version?

Reply:
It is an important point, and I think it should be right up there near the top, but I guess I just think the summary sentence should describe what it is without using a recursive definition (e.g. If all you see is "A [Foo] is a more general version of a [Bar]": if you don't know what a Bar is, it doesn't really help you).

Also, I don't like defining the "base class" in terms of its "subclasses" (even if it's aggregation, not inheritance): It feels the wrong way around.